File: python/screen.py
from typing import List, Tuple

class Screen:
    def __init__(self, height: int, width: int):
        self.height = height
        self.width = int(width / 2)

    def draw(self, cells: List[bool]):
        assert len(cells) == self.size_1d, "cells list size does not match screen size"
        for i in range(self.height):
            for j in range(self.width):
                match cells[(i*self.width)+j]:
                    case True:
                        print("██", end="")
                    case False:
                        print("  ", end="")
            print("")

    @property
    def size_1d(self) -> int:
        return self.height * self.width

    @property
    def size_2d(self) -> Tuple[int, int]:
        return self.height, self.width

File: csharp/Screen.cs
namespace GameOfLife {
    internal class Screen {
        public int height { get; }
        public int width { get; }

        public int size1D { get {
            return this.height * this.width;
        }}
        public Screen(int height, int width) {
            this.height = height;
            this.width = width;
        }

        public void Draw(IList<bool> cells) {
            for (var i = 0; i < this.height; i++) {
                for (var j = 0; j < this.width; j++) {
                    if (cells[(i * this.width) + j]) {
                        Console.Write("██");
                    }
                    else {
                        Console.Write("  ");
                    }
                }
                Console.WriteLine();
            }
        }
    }
}